compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -459,6 +469,7 @@ public void visit(BLangSimpleVariable varNode) {
         SymbolEnv varInitEnv = SymbolEnv.createVarInitEnv(varNode, env, varNode.symbol);
 
         typeChecker.checkExpr(rhsExpr, varInitEnv, lhsType);
+        types.validateNullLiteralUsage(rhsExpr, lhsType);
```

Comment:
Instead of calling this everywhere, can't we simply do this check at `visit(BLangLiteral)` method in type checker?

Reply:
Would be ideal if we can do that. But how can we identify the context `null` is being used if we do the check in `visit(BLangLiteral)` method?

Reply:
done with f0b9e85